pkg/gadgets/bindsnoop/tracer/core/tracer.go

```diff
@@ -0,0 +1,298 @@
+//go:build linux
+// +build linux
+
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019-2022 The Inspektor Gadget authors
+
+package tracer
+
+// #include <linux/types.h>
+// #include "./bpf/bindsnoop.h"
+// #include <arpa/inet.h>
+// #include <stdlib.h>
+//
+//char *ip_to_string(const struct bind_event *event) {
+//	socklen_t size;
+//	int ip_type;
+//	char *ip;
+//
+//	size = sizeof *ip * INET6_ADDRSTRLEN;
+//	ip = malloc(size);
+//  if (ip == NULL)
+//		return NULL;
+//
+//	if (event->ver == 4)
+//		ip_type = AF_INET;
+//	else if (event->ver == 6)
+//		ip_type = AF_INET6;
+//	else
+//		return NULL;
```

Comment:
Move this before calling `malloc()`, otherwise you'll leak the pointer then doing `return NULL`

Reply:
Good catch, thank you!